docs/README.md

```diff
@@ -0,0 +1,272 @@
+
+### Doobie
+
+The [Doobie](https://github.com/tpolecat/doobie) integration allows you to use the `UUID` type mapped to your database's UUID type.
+
+```scala
+libraryDependencies += "com.47deg" % "memeid-doobie" % "@VERSION@"
+```
+
+To have the [UUID mappings](https://tpolecat.github.io/doobie/docs/12-Custom-Mappings.html) available in scope you can import `memeid.doobie.implicits`.
+
+```scala mdoc
+import cats.effect._
+
+import doobie._
+import doobie.implicits._
+import doobie.h2.implicits._
+
+import memeid.doobie.implicits._
+
+import scala.concurrent.ExecutionContext
+
+implicit val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+val transactor: Transactor[IO] =
+  Transactor.fromDriverManager[IO](
+    driver = "org.h2.Driver",
+    url = "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1",
+    user = "",
+    pass = ""
+  )
+
+// Create table
+sql"CREATE TABLE IF NOT EXISTS test (id UUID NOT NULL)".update.run.transact(transactor).unsafeRunSync
+
+def select(uuid: UUID): Query0[UUID] =
+  sql"""SELECT id from test where id = ${uuid}""".query[UUID]
+
+def insert(uuid: UUID): Update0 =
+  sql"""insert into test (id) values ($uuid)""".update
+
+val example = UUID.V1.next
+
+val program: IO[UUID] = for {
+  _ <- insert(example).run.transact(transactor)
+  u <- select(example).unique.transact(transactor)
+} yield u
+
+program.unsafeRunSync
+```
+
+### Circe
+
+```scala
+libraryDependencies += "com.47deg" % "memeid-circe" % "@VERSION@"
+```
+
+You can import `memeid.circe.implicits` to have the `Encoder` and `Decoder` instances for `UUID` in scope.
+
+```scala mdoc
+import io.circe.{ Json, Encoder, Decoder }
+
+import memeid.UUID
+import memeid.circe.implicits._
+
+val uuid = UUID.V1.next
+val json = Json.fromString(uuid.toString)
+
+Encoder[UUID].apply(uuid)
+Decoder[UUID].decodeJson(json)
+```
+
+### Http4s
+
+```scala
+libraryDependencies += "com.47deg" % "memeid-http4s" % "@VERSION@"
+```
+
+### Path parameters
+
+Using `UUID` companion object we can extract UUIDs from path parameters in URLs:
+
+```scala mdoc
+import cats.effect._
+
+import org.http4s._
+import org.http4s.dsl.io._
+
+import memeid.UUID
+
+HttpRoutes.of[IO] {
+  case GET -> Root / "user" / UUID(uuid) => Ok(s"Hello, ${uuid}!")
+}
+```
+
+### Query parameters
+
+The http4s integrations provides implicit instances for `QueryParamDecoder[UUID]` and `QueryParamEncoder[UUID]`, which you can use to derive matchers for query parameters or send UUID in request query parameters.
+
+```scala mdoc
+import cats.effect._
+
+import org.http4s._
+import org.http4s.dsl.io._
+
+import memeid.UUID
+import memeid.http4s.implicits._
+
+object UUIDParamDecoder extends QueryParamDecoderMatcher[UUID]("uuid")
+
+HttpRoutes.of[IO] {
+  case GET -> Root / "user" :? UUIDParamDecoder(uuid) => Ok(s"Hello, ${uuid}!")
+}
+```
+
+## Cats & Cats-effect
```

Comment:
I would make this integration the first one in the list, since the rest of them depend on it

Reply:
I think is fine in the end, is the least useful integration imho.

Reply:
okey dokey :)